libraries/AP_HAL_Linux/GPIO_RPI.cpp

```diff
@@ -163,12 +163,15 @@ void GPIO_RPI::init()
     const int rpi_version = UtilRPI::from(hal.util)->get_rpi_version();
 
     GPIO_RPI::Address peripheral_base;
-    if(rpi_version == 1) {
+    if(rpi_version == 0) {
         peripheral_base = Address::BCM2708_PERIPHERAL_BASE;
-    } else if (rpi_version == 2) {
+    } else if (rpi_version == 1 || rpi_version == 2) {
         peripheral_base = Address::BCM2709_PERIPHERAL_BASE;
-    } else {
+    } else if (rpi_version == 3) {
         peripheral_base = Address::BCM2711_PERIPHERAL_BASE;
+    } else {
```

Comment:
I think long term we should just report the soc version. No need to decode it to 0, 1, 2... But this is an easier fix for now.